_posts/2024-11-26-opensearch-performance-2.17.md

```diff
@@ -0,0 +1,276 @@
+
+These achievements are made possible through collaboration with the broader OpenSearch community, whose contributions in testing, feedback, and development have been invaluable. Together, we are building a robust and efficient search and analytics engine capable of addressing current and future challenges.
+
+Stay connected to our [_blog_](https://opensearch.org/blog) and [_GitHub_](https://github.com/opensearch-project/OpenSearch) for ongoing updates and insights as we continue this journey of innovation and share future plans.
+
+## Appendix: Benchmarking tests and results
+
+This section outlines the performance benchmarks and results achieved using OpenSearch Benchmark to evaluate the Big5 workload across various OpenSearch versions and configurations. 
+
+### Benchmark Setup
+
+- **Benchmarking tool**: OpenSearch Benchmark, our standard benchmarking tool used in prior evaluations.  
+- **Instance type**: *c5.2xlarge* (8 vCPU, 16 GB RAM), chosen as a mid-tier option to avoid masking resource efficiency gains with oversized instances.  
+- **Cluster configuration**: Single-node cluster for reproducibility and ease of setup.  
+- **Index setup**: Big5 Index configured with one shard and no replicas (`--workload-params="number_of_replicas:0"`).  
+- **Corpus details**: 100 GB dataset with 116 million documents. The storage size after ingestion was 24 GB for the primary shard.  After removing overheads like doc values, the RSS is expected to be around 8 GB, matching the JVM heap size for the instance. Keeping most of the data in memory ensures a more accurate evaluation of performance improvements.
+- **Ingestion**: Conducted with a single bulk indexing client to ensure that data is ingested in chronological order.
+- **Merge policy**:  
+  - LogByteSizeMergePolicy for OpenSearch 2.11.0 and later.  
+  - TieredMergePolicy for OpenSearch 1.3.18 and 2.7.0.  
+- **Test mode**: Tests were run in [benchmarking mode](https://opensearch.org/docs/latest/benchmark/user-guide/target-throughput/#benchmarking-mode) (`target-throughput` disabled) so that the OpenSearch client sent requests to the OpenSearch cluster as fast as possible. 
+
+### Key notes and considerations
+
+- **Additional queries**: The Big5 workload was recently updated to include additional queries. These queries have been incorporated in the results of this blog. 
+- **`multi_terms-keyword` support**:  OpenSearch 1.3.18 and 2.7.0 recorded `0` ms service time for `multi_terms-keyword`. This is because `multi_terms-keyword` was not supported until OS 2.11.0. Thus, entries in **Mean latency** account for this by excluding `multi_terms-keyword` from the geometric mean computation for OS 1.3.18 and OS 2.7.  
```

Comment:
```suggestion
- **`multi_terms-keyword` support**: OpenSearch 1.3.18 and 2.7.0 recorded `0` ms service time for `multi_terms-keyword`. This is because `multi_terms-keyword` was not supported until OpenSearch 2.11.0. Thus, entries in **Mean latency** account for this by excluding `multi_terms-keyword` from the geometric mean computation for OpenSearch 1.3.18 and 2.7.0.  
```